modules/react/common/spec/responsiveContext.spec.tsx

```diff
@@ -0,0 +1,58 @@
+import React from 'react';
+import useResizeObserver from 'use-resize-observer';
+import {Box, Flex} from '../../layout';
+import {
+  useResponsiveContext,
+  isWithinBreakpoint,
+  ResponsiveContextProvider,
+} from '../lib/responsive/responsiveContext';
+import {breakpoints} from '../lib/theming';
+const context = describe;
+
+describe('Within Breakpoint function', () => {
+  context('when using the isWithinBreakpoint function', () => {
+    it('should return true if width is within breakpoints', () => {
+      const withinBreakpoint = isWithinBreakpoint(1000, breakpoints.m, breakpoints.l);
+
+      expect(withinBreakpoint).toBe(true);
+    });
+
+    it('should return false if width is not within breakpoints', () => {
+      const withinBreakpoint = isWithinBreakpoint(500, breakpoints.m, breakpoints.l);
+
+      expect(withinBreakpoint).toBe(false);
+    });
+  });
```

Comment:
1. Only give a min with a width that's under the breakpoint
2. Only give a min with a width that's over the breakpoint
3. Give a min and max with a width that's under the breakpoint range
4. Give a min and max with a width that's in the breakpoint range
5. Give a min and max with a width that's above the breakpoint range
6. Give a negative width and any breakpoint

Reply:
Done